sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -397,7 +408,7 @@ object IntervalUtils {
       secondStr: String,
       sign: Int): Long = {
     var micros = 0L
-    val days = toLongWithRange(DAY, dayStr, 0, Int.MaxValue).toInt
+    val days = toLongWithRange(DAY, dayStr, 0, MAX_DAY).toInt
```

Comment:
Do we have tests for MAX_DAY + 1 and MAX_DAY, and for MIN_DAY-1 and MIN_DAY?

Reply:
Added in interval.sql